java-api-core/src/com/kinvey/java/sync/SyncManager.java

```diff
@@ -144,7 +144,10 @@ public void enqueueRequest(String collectionName, AbstractKinveyJsonClientReques
         entityQuery.equals(META_DOT_ID, id);
         List<SyncItem> itemsList = requestCache.get(entityQuery);
 
-        if (itemsList.size() == 0) {
+        if (itemsList.isEmpty()) {
+            requestCache.save(createSyncItem(collectionName, method, networkManager, id));
+        } else if (method == RequestMethod.DELETE) {
+            requestCache.delete(entityQuery);
             requestCache.save(createSyncItem(collectionName, method, networkManager, id));
```

Comment:
Does similar logic have to be implemented for `public <T extends GenericJson> void enqueueRequests(...)` methods?

Reply:
Yes, it has to be implemented for `enqueueRequests` as well. Updated.